docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,271 @@
+repoint some important data onto the peristent volume.
+
+By default we use aufs over ext4.  `DOCKER_STORAGE=btrfs` is also a good choice
+for a filesystem: it is relatively reliable with Docker; btrfs itself is much
+more reliable than it used to be with modern kernels.  It can easily span
+multiple volumes, which is particularly useful when we are using an instance
+type with multiple ephemeral instance disks.
+
+### AutoScaling
+
+We run the minions in an AutoScalingGroup.  Currently auto-scaling (e.g. based
+on CPU) is not actually enabled (#11935).  Instead, the auto-scaling group
+means that AWS will relaunch any minions that are terminated.
+
+We do not currently run the master in an AutoScalingGroup, but we should
+(#11934)
+
+### Networking
+
+Kubernetes uses an IP-per-pod model.  This means that a node, which runs many
+pods, must have many IPs.  The way we implement this on AWS is to use VPCs and
+the advanced routing support that it allows.  Each pod is assigned a /24 CIDR;
+then this CIDR is configured to route to an instance in the VPC routing table.
+
+It is also possible to use overlay networking on AWS, but the default kube-up
+configuration does not.
+
+### NodePort & LoadBalancing
+
+Kubernetes on AWS integrates with ELB.  When you create a service with
+Type=LoadBalancer, kubernetes (the kube-controller-manager) will create an ELB,
+create a security group for the ELB which allows access on the service ports,
+attach all the minions to the ELB, and modify the security group for the
+minions to allow traffic from the ELB to the minions.  This traffic reaches
+kube-proxy where it is then forwarded to the pods.
+
+ELB requires that all minions listen on a single port, and it acts as a layer-7
+forwarding proxy (i.e. the source IP is not preserved).  It is not trivial for
+kube-proxy to recognize the traffic therefore.  So, LoadBalancer services are
+also exposed as NodePort services.  For NodePort services, a cluster-wide port
+is assigned by kubernetes to the service, and kube-proxy listens externally on
+that port on every minion, and forwards traffic to the pods.  So for a
+load-balanced service, ELB is configured to proxy traffic on the public port
+(e.g. port 80) to the NodePort assigned to the service (e.g. 31234), kube-proxy
+recognizes the traffic coming to the NodePort by the inbound port number, and
+send it to the correct pods for the service.
+
+Note that we do not automatically open NodePort services in the AWS firewall
+(although we do open LoadBalancer services).  This is because we expect that
+NodePort services are more of a building block for things like inter-cluster
+services or for LoadBalancer.  To consume a NodePort service externally, you
+will likely have to open the port in the minion security group
+(`kubernetes-minion-<clusterid>`).
+
+### IAM
+
+kube-proxy sets up two IAM roles, one for the master called
+(kubernetes-master)[cluster/aws/templates/iam/kubernetes-master-policy.json]
```

Comment:
Preview the doc. Link syntax is backward. Also please use working relative paths: ../../cluster/...